### PR TITLE
Conditional rendering in navbar, protected graphql fields, and feedback on login/signup error

### LINKED ISSUE #38: Implement conditional rendering for Login/Signup in nav bar

### LINKED ISSUE #68: Add functionality to protect certain graphql fields from being by users not authenticated
The user auth status will be included in the context argument in the resolver. Return null from the fields unauthenticated users should not be able to query.

### LINKED ISSUE #72: Provide feedback for user on invalid login/signup

### PR DESCRIPTION
**Pull request by evanmcneely**
### Summary

1. Conditionally rendering login/signup or dashboard/logout based on auth status in navbar
2. Added protection based on auth status on graphql fields 
3. Giving feedback to the user on unsuccessful login or signup

### Type of Change

-   [ ] New feature (non-breaking change which adds functionality)

### Issues

closes #38 
closes #72 
closes #68

### CODE REVIEW

**Review by feiw101 — COMMENTED** [review]
Awesome!! No requests on change. Will approve this, once the previous PRs are approved!

**Review by donjewey — APPROVED** [review]

**Review by feiw101 — APPROVED** [review]

**Pull request merged**
